test/groovy/util/JenkinsReadJsonRule.groovy

```diff
@@ -31,8 +31,13 @@ class JenkinsReadJsonRule implements TestRule {
                         return js.parseText(m.text)
                     } else if(m.file) {
                         def js = new JsonSlurper()
-                        def reader = new BufferedReader(new FileReader( "${this.testRoot}${m.file}" ))
-                        return js.parse(reader)
+                        if (testInstance.binding.hasVariable('files')) {
```

Comment:
Wouldn't it be better not to have the test root into the game when having the files stored in memory. Having a test root is only needed when we persist files to disk in a temporary directory.